src/spikeinterface/core/tests/test_generate.py

```diff
@@ -33,134 +59,87 @@ def measure_memory_allocation(measure_in_process: bool = True) -> float:
-    )
-    assert ratio <= 1.0 + relative_tolerance, assertion_msg
-
-    traces = lazy_recording.get_traces()
-    expected_traces_shape = (int(durations[0] * sampling_frequency), num_channels)
-
-    traces_size_MiB = traces.nbytes / bytes_to_MiB_factor
-    assert traces_size_MiB == expected_trace_size_MiB
-    assert traces.shape == expected_traces_shape
-
-    memory_after_traces_MiB = measure_memory_allocation() / bytes_to_MiB_factor
-
-    expected_memory_usage_MiB = memory_after_instanciation_MiB + traces_size_MiB
-    ratio = memory_after_traces_MiB * 1.0 / expected_memory_usage_MiB
-    assertion_msg = (
-        f"Memory after loading traces is {memory_after_traces_MiB} MiB and is {ratio:.2f} times"
-        f"the expected memory usage of {expected_memory_usage_MiB} MiB."
+        strategy="tile_pregenerated",
+        noise_block_size=noise_block_size,
     )
-    assert ratio <= 1.0 + relative_tolerance, assertion_msg
-
-
-@pytest.mark.parametrize("mode", mode_list)
-def test_generate_lazy_recording(mode):
-    # Test that get_traces does not consume more memory than allocated.
-    bytes_to_MiB_factor = 1024**2
-    full_traces_size_GiB = 1.0
-    relative_tolerance = 0.05  # relative tolerance of 5 per cent
-
-    initial_memory_MiB = measure_memory_allocation() / bytes_to_MiB_factor
-
-    lazy_recording = generate_lazy_recording(full_traces_size_GiB=full_traces_size_GiB, mode=mode)
-
-    memory_after_instanciation_MiB = measure_memory_allocation() / bytes_to_MiB_factor
-    expected_memory_usage_MiB = initial_memory_MiB
-    if mode == "white_noise":
-        expected_memory_usage_MiB += 50  # 50 MiB for the white noise generator
-
-    ratio = memory_after_instanciation_MiB * 1.0 / expected_memory_usage_MiB
-    assertion_msg = (
-        f"Memory after instantation is {memory_after_instanciation_MiB} MiB and is {ratio:.2f} times"
-        f"the expected memory usage of {expected_memory_usage_MiB} MiB."
-    )
-    assert ratio <= 1.0 + relative_tolerance, assertion_msg
-
-    traces = lazy_recording.get_traces()
-    traces_size_MiB = traces.nbytes / bytes_to_MiB_factor
-    assert full_traces_size_GiB * 1024 == traces_size_MiB
-
-    memory_after_traces_MiB = measure_memory_allocation() / bytes_to_MiB_factor
-
-    expected_memory_usage_MiB = memory_after_instanciation_MiB + traces_size_MiB
-    ratio = memory_after_traces_MiB * 1.0 / expected_memory_usage_MiB
-    assertion_msg = (
-        f"Memory after loading traces is {memory_after_traces_MiB} MiB and is {ratio:.2f} times"
-        f"the expected memory usage of {expected_memory_usage_MiB} MiB."
+    after_instanciation_MiB = measure_memory_allocation() / bytes_to_MiB_factor
+    memory_usage_MiB = after_instanciation_MiB - before_instanciation_MiB
+    expected_allocation_MiB = dtype.itemsize * num_channels * noise_block_size / bytes_to_MiB_factor
+    ratio = expected_allocation_MiB / expected_allocation_MiB
```

Comment:
@samuelgarcia You modified this test so it always passes? : P

Reply:
hé hé. I see it now.
this is some kind of a mistake.

Reply:
Let's fix it later : D